hangman.py

```diff
@@ -152,62 +253,118 @@ def show_possible_matches(my_word):
+    guesses = 0
+    warnings = 0
+    unique_letters = ''
+    for letter in secret_word:
+        if letter not in unique_letters:
+            unique_letters += letter
+
+    print('Welcome to the game Hangman!')
+    print('I\'m thinking of a word that is ' + str(len(secret_word)) + ' letters long.')
+    print('-----------------')
+
+    while guesses < 6:
+        letters_left = get_available_letters(letters_guessed)
+        print('You have ' + str(6 - guesses) + ' guesses left.')
+        print('Available letters: ' + letters_left)
+
+        user_guess = (input('Please guess a letter: '))
+
+        if str.isalpha(user_guess):
+            if str.isupper(user_guess):
+                user_guess = str.lower(user_guess)
+            if user_guess in letters_guessed:
+                warnings += 1
+                if warnings < 4:
+                    print('Oops! You\'ve already guessed that letter.You now have ' + str(warnings)
+                          + 'left:' + get_guessed_word(secret_word, letters_guessed))
+                elif warnings == 4:
+                    print('Oops! You\'ve already guessed that letter.You now have no warnings left:\n '
+                          'so you lose one guess:' + get_guessed_word(secret_word, letters_guessed))
+                    guesses += 1
+                else:
+                    break
+
+            else:
+                letters_guessed.append(user_guess)
+                if is_word_guessed(user_guess, secret_word):
+                    print('Good guess:', get_guessed_word(secret_word, letters_guessed))
+                    print('--------')
+                    if get_guessed_word(secret_word, letters_guessed) == secret_word:
+                        print('------------\nCongratulations, you won!\n'
+                              'Your total score for this game is:' + str((len(set(secret_word)) * (6 - guesses))))
+                        break
+
+                else:
+                    if user_guess in ['a', 'e', 'i', 'o', 'u']:
+                        print('Oops! That letter is not in my word:' + get_guessed_word(secret_word, letters_guessed))
+                        guesses += 2
+                    else:
+                        print('Oops! That letter is not in my word:\n'
+                              'Please guess a letter:' + get_guessed_word(secret_word, letters_guessed))
+                        guesses += 1
+
+        elif user_guess == '*':
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+
+        else:
+            warnings += 1
+            if warnings < 3:
+                print('Oops! That is not a valid letter. You have ' + str(warnings) + ' warnings left:' +
+                      get_guessed_word(secret_word, letters_guessed))
+            else:
+                break
+
+    if guesses == 6:
+        print('-----------\n'
+              'Sorry, you ran out of guesses.'
+              ' The word was' + secret_word)
```

Comment:
Can you split the `secret_word` from the string. It shows: '...The word wasyourword'